pkg/referrerstore/oras/authprovider/authprovider.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package authprovider
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/containerd/containerd/reference"
+	"github.com/docker/cli/cli/config"
+	"github.com/docker/cli/cli/config/configfile"
+)
+
+// This config represents the credentials that should be used
+// when pulling artifacts from specific repositories.
+type AuthConfig struct {
+	Username string
+	Password string
+	Email    string
+	Provider AuthProvider
+}
+
+type AuthProvider interface {
+	// Enabled returns true if the config provider is properly enabled
+	// It will verify necessary values provided in config file to
+	// create the AuthProvider
+	Enabled() bool
+	// Provide returns AuthConfig for registry.
+	Provide(artifact string) (AuthConfig, error)
+}
+
+type defaultProviderFactory struct{}
+type defaultAuthProvider struct {
+	configPath string
+}
+
+type defaultAuthProviderConf struct {
+	Name       string `json:"name"`
+	ConfigPath string `json:"configPath,omitempty"`
+}
+
+const DefaultAuthProviderName string = "docker-config"
```

Comment:
Isn't the default auth provider provided by the oras-go library already?

Reply:
Yes I believe this is true for the current version of the oras-go we leverage. However, for v2, I currently don't see built in support for docker config (maybe it'll be added later?). This is why I added the dedicated docker config support for the default provider.

Reply:
Can we confirm if this is in the roadmap of oras which I would assume